kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration-audit-replication/src/main/java/org/kie/server/spring/boot/autoconfiguration/audit/replication/AuditDataReplicationKieServerAutoConfiguration.java

```diff
@@ -0,0 +1,131 @@
+
+
+    @Bean
+    @ConditionalOnMissingBean(name = "jmsSender")
+    public JMSSender createAuditDataReplicationSender() {
+        if (env.containsProperty("kieserver.audit-replication.queue")) {
+            String endpointName = env.getProperty("kieserver.audit-replication.queue");
+            return new JMSSender(endpointName);
+        } else {
+            String endpointName = env.getProperty("kieserver.audit-replication.topic");
+            return new JMSSender(endpointName);
+        }
+    }
+
+
+    @Bean
+    @ConditionalOnMissingBean(name = "xstreamBean")
+    public XStream createXStream() {
+        XStream xstream = XStreamUtils.createTrustingXStream();
+        String[] voidDeny = {"void.class", "Void.class"};
+        xstream.denyTypes(voidDeny);
+        return xstream;
+    }
+
+    @Bean(name = "auditEntityManagerFactory")
+    @ConditionalOnMissingBean(name = "auditEntityManagerFactory")
+    @ConditionalOnProperty(name = "kieserver.audit-replication.consumer", havingValue = "true")
+    public LocalContainerEntityManagerFactoryBean entityManagerFactory(DataSource dataSource, JpaProperties jpaProperties) {
+        return EntityManagerFactoryHelper.create(applicationContext,
+                                                 dataSource,
+                                                 jpaProperties,
+                                                 "org.jbpm.audit",
+                                                 "classpath:/META-INF/jbpm-audit-persistence.xml");
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(name = "auditDataReplicationProcessEventListenerProducer")
+    @ConditionalOnProperty(name = "kieserver.audit-replication.producer", havingValue = "true")
+    public ProcessEventListener createrocessEventListenerProducer() {
+        logger.info("Adding AuditDataReplicationProcessEvent from data replication");
+        return new AuditDataReplicationProcessEventProducer();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(name = "auditDataReplicationTaskLifeCycleEventListenerProducer")
+    @ConditionalOnProperty(name = "kieserver.audit-replication.producer", havingValue = "true")
+    public TaskLifeCycleEventListener createTaskLifeCycleEventListener(EntityManagerFactory emf) {
+        logger.info("Adding AuditDataReplicationTaskLifeCycleEventListenerProducer from data replication");
+        return new AuditDataReplicationTaskLifeCycleEventListenerProducer();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(name = "auditDataReplicationBAMTaskSumaryListenerProducer")
+    @ConditionalOnProperty(name = "kieserver.audit-replication.producer", havingValue = "true")
+    public TaskLifeCycleEventListener createBAMTaskSummaryListener(EntityManagerFactory emf) {
+        logger.info("Adding AuditDataReplicationBAMTaskSumaryListenerProducer from data replication");
+        return new AuditDataReplicationBAMTaskSumaryListenerProducer();
+    }
+
+    @Bean(name="auditDataReplicationQueueConsumer")
```

Comment:
This bean is more generic, so it might be either a queue or a topic, so I would rename it to 
```suggestion
    @Bean(name="auditDataReplicationConsumer")
```

Reply:
done